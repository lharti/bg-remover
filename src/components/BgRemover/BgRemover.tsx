'use client'

import { DropZone } from '@/components/DropZone'
import { ImagesList } from '@/components/ImagesList'
import { cn } from '@/utils/cn'
import React, { useState } from 'react'

export interface BgRemoverProps {
    className?: string
}

export const BgRemover: React.FC<BgRemoverProps> = ({ className }) => {
    const [files, setFiles] = useState<File[]>([])

    const onDrop = (acceptedFiles: File[]) => {
        setFiles(prevFiles => prevFiles.concat(acceptedFiles))
    }

    return (
        <div className={cn('space-y-10', className)}>
            <DropZone className="mx-auto max-w-xl" onDrop={onDrop} />

            <ImagesList files={files} className="mx-auto" />
        </div>
    )
}
