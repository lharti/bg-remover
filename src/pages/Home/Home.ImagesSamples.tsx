import { SAMPLES } from '@/pages/Home/Home.constants'
import React from 'react'

interface ImagesSamplesProps {
    onSampleClick: (file: File) => void
}
export const ImagesSamples: React.FC<ImagesSamplesProps> = ({
    onSampleClick: onSampleClick,
}) => {
    const loadSample = async (url: string) => {
        const response = await fetch(url)
        const blob = await response.blob()

        const file = new File([blob], 'sample.jpg', {
            type: 'image/jpeg',
        })

        onSampleClick(file)
    }

    return (
        <div className="mt-10 space-y-4 text-center">
            <p className="font-bold">No images? Try one of these:</p>

            <div className="flex justify-center gap-4">
                {SAMPLES.map(({ name, thumbnail, url }) => (
                    <button
                        key={name}
                        className="hover:opacity-80"
                        role="button"
                        aria-label={`Select sample image: ${name}`}
                        onClick={() => loadSample(url)}
                    >
                        <img
                            src={thumbnail}
                            alt={name}
                            className={`w-40 rounded-lg object-cover shadow-lg`}
                        />
                    </button>
                ))}
            </div>
        </div>
    )
}
