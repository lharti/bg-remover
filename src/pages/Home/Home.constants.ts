export const ASSETS_CDN = 'https://assets.lharti.com/bg'

export const SAMPLES = [
    {
        name: 'person',
        thumbnail: `${ASSETS_CDN}/samples/sm/person.jpg`,
        url: `${ASSETS_CDN}/samples/lg/person.jpg`,
    },

    {
        name: 'object',
        thumbnail: `${ASSETS_CDN}/samples/sm/object.jpg`,
        url: `${ASSETS_CDN}/samples/lg/object.jpg`,
    },

    {
        name: 'animal',
        thumbnail: `${ASSETS_CDN}/samples/sm/animal.jpg`,
        url: `${ASSETS_CDN}/samples/lg/animal.jpg`,
    },
]
