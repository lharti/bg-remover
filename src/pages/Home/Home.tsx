import { DropZone } from '@/components/ui/DropZone'
import { generateImagesItemsFromFiles } from '@/libs/generateImageItemsFromFiles'
import { ImageItem } from '@/types/imageItem'
import React from 'react'
import { ImagesSamples } from './Home.ImagesSamples'

export interface HomePageProps {
    onImagesSelect: (files: ImageItem[]) => void
}

export const HomePage: React.FC<HomePageProps> = ({ onImagesSelect }) => {
    const addItemsFromFiles = (files: File[]) => {
        const imageItems = generateImagesItemsFromFiles(files)

        onImagesSelect(imageItems)
    }

    const onSampleClick = (file: File) => addItemsFromFiles([file])

    return (
        <div
            className={`
              mx-auto mt-16 flex max-w-2xl flex-col items-center justify-center
              space-y-12 px-4
            `}
        >
            <header className="text-center">
                <h1 className="text-4xl font-bold">AI Background Remover</h1>

                <p className="mt-4 text-lg">
                    Upload your images and remove backgrounds automatically
                </p>
            </header>

            <DropZone className="w-full py-10" onDrop={addItemsFromFiles} />

            <ImagesSamples onSampleClick={onSampleClick} />
        </div>
    )
}
