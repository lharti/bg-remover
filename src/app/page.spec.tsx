import { BgRemover } from '@/components/BgRemover'
import { render } from '@testing-library/react'
import React from 'react'
import Home from './page'

jest.mock('@/components/BgRemover')

jest.mocked(BgRemover).mockImplementation(({ className }) => (
    <div className={className}>{'BgRemover'}</div>
))

describe('<Home />', () => {
    it('should render', () => {
        expect.assertions(1)

        const { container } = render(<Home />)

        expect(container).toMatchInlineSnapshot(`
            <div>
              <div
                class="mx-auto mt-20 max-w-screen-lg px-6"
              >
                BgRemover
              </div>
            </div>
        `)
    })
})
