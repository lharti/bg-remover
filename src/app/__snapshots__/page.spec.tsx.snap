// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render 1`] = `
<div>
  <div>
    Hello
  </div>
</div>
`;
